packages/table/src/hooks/useTableMap.js

```diff
@@ -0,0 +1,43 @@
+import React, { useCallback } from "react";
+
+import { TR, TD } from "../components.styled";
+
+const defaultCellComponents = {};
+const defaultPlaceholders = {};
+
+export default ({
+  columns,
+  keyProp,
+  rowComponent: Row = TR,
+  cellComponents = defaultCellComponents,
+  placeholders = defaultPlaceholders,
+} = {}) => {
+  return useCallback(
+    (row, index) => (
+      <Row key={row[keyProp] || index} $rowData={row} $rowIndex={index}>
+        {columns.map((columnName, columnIndex) => {
+          const Cell =
+            cellComponents[columnName] || cellComponents.$default || TD;
+
+          const content =
+            row[columnName] ||
+            placeholders[columnName] ||
+            placeholders.$default;
+
+          return (
+            <Cell
+              key={columnName}
```

Comment:
This `key` will repeat on every row... is it intended? 🤔

Reply:
It won't be a problem because the rows have their own keys (which should be unique).